Eval.hs

```diff
@@ -444,6 +452,22 @@ isIdPair :: Val -> Bool
 isIdPair VIdPair{} = True
 isIdPair _         = False
 
+idComp :: Val -> Val -> Val -> Val -> Val -> Val -> Val
+idComp a u v w pId qId = case (pId, qId) of
+  (VIdPair p ps, VIdPair q qs)          ->
+    let i:j:_ = freshs [a,u,v,w,pId,qId] in
+    let
+      rs = unionSystem
+        (unionSystem
+          (border (q @@ (Atom i :/\: Atom j)) ps)
+          (border (p @@ (Atom i :\/: NegAtom j)) qs))
+        (mkSystem [((i ~> 0), p @@ NegAtom j), ((i ~> 1), q @@ Atom j)])
+    in
+    VIdPair (VPLam i (comp j a v rs)) (joinSystem (border ps qs))
+  (VIdPair p ps, qId) | eps `member` ps -> qId
+  (pId, VIdPair q qs) | eps `member` qs -> pId
```

Comment:
As some justification, imagine that face formulae were first class objects and we could see that Id types were pairs of a path and a face formula. Then, using eta for pairs and paths, we would have:

```
idComp (<i> a , 1) qId
   = idComp (<i> a , 1) (qId.1 , qId.2)
   = (<i> comp j A [ 1 -> qId.1 @ (i /\ j) , qId.2 \/ (i=0) -> a ] a , 1 /\ qId.2)
   = (<i> qId.1 @ i , qId.2)
   = (qId.1 , qId.2)
   = qId
```